pkg/types/chain_writer.go

```diff
@@ -7,6 +7,10 @@ import (
 	"github.com/smartcontractkit/chainlink-common/pkg/services"
 )
 
+const (
+	ErrSettingTransactionGasLimitNotSupported = UnimplementedError("setting transaction gas limit is not supported")
```

Comment:
What are the differences between `Unimplemented` and `InvalidArgument`?
```suggestion
	ErrSettingTransactionGasLimitNotSupported = InvalidArgumentError("setting transaction gas limit is not supported")
```

Reply:
was 50/50 on this, seemed like either could apply, I've update to InvalidArgumentError